### PR TITLE
Optionally split out the pushers into a separate process

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: synapse/app/pusher.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+from synapse.storage import DataStore
+from synapse.util.async import sleep
+from synapse.util.logcontext import (LoggingContext, preserve_fn)
+
+from twisted.internet import reactor, defer
+
+import sys
+import logging
+
+logger = logging.getLogger("synapse.app.pusher")
```
Comment (erikjohnston) [inline synapse/app/pusher.py]: Why not `__name__`?

Comment (NegativeMjark) [inline synapse/app/pusher.py]: It might be running as ``__main__``

Comment (NegativeMjark) [inline synapse/app/pusher.py]: Which I think is the reason that ``synapse.app.homeserver`` does the same thing.

**Comment by erikjohnston** [comment]
LGTM?

**Pull request merged**